src/main/java/mate/academy/security/jwt/JwtTokenProvider.java

```diff
@@ -0,0 +1,86 @@
+package mate.academy.security.jwt;
+
+import io.jsonwebtoken.Claims;
+import io.jsonwebtoken.Jws;
+import io.jsonwebtoken.JwtException;
+import io.jsonwebtoken.Jwts;
+import io.jsonwebtoken.SignatureAlgorithm;
+import io.jsonwebtoken.security.Keys;
+import java.security.Key;
+import java.util.Date;
+import java.util.List;
+import javax.annotation.PostConstruct;
+import javax.servlet.http.HttpServletRequest;
+import mate.academy.exception.InvalidJwtAuthenticationException;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.stereotype.Component;
+
+@Component
+public class JwtTokenProvider {
+    private static final int TOKEN_START_INDEX = 7;
+    private final UserDetailsService userDetailsService;
+    @Value("${jwt.token.validity-time}")
+    private long validityTime;
+    private Key jwtSecretKey;
+
+    public JwtTokenProvider(UserDetailsService userDetailsService) {
+        this.userDetailsService = userDetailsService;
+    }
+
+    @PostConstruct
+    private void init() {
+        jwtSecretKey = Keys.secretKeyFor(SignatureAlgorithm.HS256);
+    }
+
+    public String generateToken(String login, List<String> roles) {
+        Date now = new Date();
+        Date expirationDate = new Date(now.getTime() + validityTime);
+        Claims claims = Jwts.claims().setSubject(login);
+        claims.put("roles", roles);
+        return Jwts.builder()
+                .setClaims(claims)
+                .setIssuedAt(now)
+                .setExpiration(expirationDate)
+                .signWith(jwtSecretKey, SignatureAlgorithm.HS256)
```

Comment:
```suggestion
                .signWith(SignatureAlgorithm.HS256, jwtSecretKey)
```

Reply:
This variant is deprecated, the official documentation recommends that it should not be used.

![image](https://user-images.githubusercontent.com/101512791/186399676-b17148e7-236a-4aca-884a-912f948025fc.png)